lib/rubocop/cop/util.rb

```diff
@@ -121,17 +121,25 @@ def to_supported_styles(enforced_style)
           .sub('Style', 'Styles')
       end
 
-      def tokens(node)
+      def tokens(node) # rubocop:disable Metrics/AbcSize
         @tokens ||= {}
         return @tokens[node.object_id] if @tokens[node.object_id]
 
         source_range = node.source_range
         begin_pos = source_range.begin_pos
         end_pos = source_range.end_pos
 
-        @tokens[node.object_id] = processed_source.tokens.select do |token|
-          token.end_pos <= end_pos && token.begin_pos >= begin_pos
+        tokens_to_node_end = processed_source.tokens.take_while do |token|
+          token.end_pos <= end_pos
         end
+
+        node_tokens = []
+        tokens_to_node_end.reverse_each do |token|
+          break unless token.begin_pos >= begin_pos
+          node_tokens.unshift(token)
```

Comment:
`reverse_each` does not create a new array in reverse order. Instead, it starts at the end of the collection and works backwards.

Once it works back past the beginning of the node, the loop breaks. This means many less calculations of `token.begin_pos` and `>=`.